### PR TITLE
Add an assertion to limit allocated bytes for current thread

### PR DESCRIPTION
**Pull request by mfussenegger**
## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java [hunk 1]
```diff
@@ -60,16 +56,17 @@
...
+            .setTableStats(tableStats)
             .addView(new RelationName("doc", "v2"), "select a, x from doc.t1")
             .addView(new RelationName("doc", "v3"), "select a, x from doc.t1")
             .build();
-        tableStats = new TableStats();
     }
 
     private LogicalPlan plan(String statement) {
-        return plan(statement, sqlExecutor, clusterService, tableStats);
+        return withMemoryLimit(ByteSizeUnit.MB.toBytes(8), () -> sqlExecutor.logicalPlan(statement));
```
Comment (seut) [inline sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java]: Why is this limited to 8MB?

Comment (mfussenegger) [reply sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java]: Currently the plan method allocates around 5-7 MB. The intention here is to have a regression check to make sure we don't increase the allocations a lot and also to make us aware of what various methods are allocating. For example, the deleted block below caused an extra ~40MB to be allocated: (probably due to `getFunctions()`)

```
PlannerContext context = sqlExecutor.getPlannerContext(clusterService.state());
        AnalyzedRelation relation = sqlExecutor.analyze(statement);
        LogicalPlanner logicalPlanner = new LogicalPlanner(
            getFunctions(),
            tableStats,
            () -> clusterService.state().nodes().getMinNodeVersion()
        );
        SubqueryPlanner subqueryPlanner = new SubqueryPlanner((s) -> logicalPlanner.planSubSelect(s, context));
```

Comment (seut) [reply sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java]: I see thanks, yes wording was a bit confusing. I'd suggest `assertMaxBytesAllocated`.

Comment (mfussenegger) [reply sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java]: @seut Changed it.

**Pull request merged**